pkg/operator/ceph/object/status.go

```diff
@@ -90,15 +91,28 @@ func updateStatus(ctx context.Context, observedGeneration int64, client client.C
 }
 
 func buildStatusInfo(cephObjectStore *cephv1.CephObjectStore) map[string]string {
+	nsName := fmt.Sprintf("%s/%s", cephObjectStore.Namespace, cephObjectStore.Name)
+
 	m := make(map[string]string)
 
-	if cephObjectStore.Spec.Gateway.SecurePort != 0 && cephObjectStore.Spec.Gateway.Port != 0 {
-		m["secureEndpoint"] = BuildDNSEndpoint(GetStableDomainName(cephObjectStore), cephObjectStore.Spec.Gateway.SecurePort, true)
+	advertiseEndpoint, err := cephObjectStore.GetAdvertiseEndpointUrl()
+	if err != nil {
+		// lots of validation happens before this point, so this should be nearly impossible
+		logger.Errorf("failed to get advertise endpoint for CephObjectStore %q to record on status; continuing without this. %v", nsName, err)
+	}
+
+	if cephObjectStore.AdvertiseEndpointIsSet() {
+		// if the advertise endpoint is explicitly set, it takes precedence as the only endpoint
+		m["endpoint"] = advertiseEndpoint
+		return m
+	}
```

Comment:
We didn't explicitly talk about this in the design, but I think this behavior/prioritization makes sense.

Reply:
So if the advertise endpoint is set, we always use it for internal ops too, right?

Reply:
No. Maybe there is confusion here I can clear up somehow.

For admin ops, the default endpoint is always used. That's the service endpoint, except for external clusters where it's the first `externalRgwEndpoints` entry.

This function sets the `status.Info["endpoint"]` (and sometimes `status.Info["secureEndpoint"]`) status fields. My thinking was that users will more likely look to these values for how they consume it, rather than evidence of internal usage.

I had considered adding `adminOpsEndpoint` to the status as well, and/or adding `advertiseEndpoint`. But I'm not sure how useful that would be.

Reply:
Ok thanks for the clarification. You probably clarified that in a previous conversation too.